variations.py

```diff
@@ -25,25 +25,28 @@ def __init__(self,
         Variation.__init__(self, name)
         self.variation = variation
 
-    def create(self, unit):
+    def create(self, unit):    
         new_selections = deepcopy(unit.selections)
         new_actions = deepcopy(unit.actions)
+
         if self.variation not in unit.dataset.quantities_per_vars:
             logger.fatal('Variation {} not found in ntuple'.format(self.variation))
             raise NameError
         else:
             list_of_quants = unit.dataset.quantities_per_vars[self.variation]
-            for quant_i in list_of_quants:
-                for cut_i in range(len(new_selections[0].cuts)):
-                    if quant_i in new_selections[0].cuts[cut_i].expression:
-                        new_selections[0].cuts[cut_i].expression = new_selections[0].cuts[cut_i].expression.replace(quant_i, "{quant}__{var}".format(quant=quant_i, var=self.variation))
-                for weight_i in range(len(new_selections[1].weights)):
-                    if quant_i in new_selections[1].weights[weight_i].expression:
-                        new_selections[1].weights[weight_i].expression = new_selections[1].weights[weight_i].expression.replace(quant_i, "{quant}__{var}".format(quant=quant_i, var=self.variation))
-            for act_i in new_actions:
-                act_i.name = act_i.name.replace("Nominal", self.name)
-
+            for quant in list_of_quants:
+                for sel_obj in new_selections:
+                    for cut in sel_obj.cuts:
+                        if quant in cut.expression:
+                            cut.expression = cut.expression.replace(quant, "{quant}__{var}".format(quant=quant, var=self.variation))
+                    for weight in sel_obj.weights:
+                        if quant in weight.expression:
+                            weight.expression = weight.expression.replace(quant, "{quant}__{var}".format(quant=quant, var=self.variation))
+                for act in new_actions:
+                    if quant in act.name:
```

Comment:
I would check for the same expression that is used in the expression, i.e. variable member, here because that is the quantity that matters in the end I think. In your implementation, you rely on the user to provide the same name and variable definition which might not necessarily be the case (even though it will be in probably 99 % of the cases).
This way also unrolled histograms should be correct. Maybe it is worth to check that this code has the correct behavior when it produces unrolled histograms, i.e. as implemented here:
https://github.com/KIT-CMS/ntuple_processor/blob/dev/run.py#L224-L230
```suggestion
                    if quant in act.variable:
```